goserver/session/session_test.go

```diff
@@ -0,0 +1,70 @@
+package session
+
+import (
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestNew(t *testing.T) {
+	sm := NewStoreManager(NewMemoryStore(), func() string { return "test-id" })
+	rr := httptest.NewRecorder()
+	r := httptest.NewRequest("method", "https://test.com", nil)
+	sess, err := New(sm, rr, r)
+	if err != nil {
+		t.Errorf("cannot create new session, got err: %v", err)
+	}
+
+	// Make sure session is created in manager.
+	sess, err = sm.Find(sess.ID)
+	if err != nil {
+		t.Fatalf("cannot find session in session manager, got err: %v", err)
+	}
+	if sess == nil {
+		t.Fatalf("cannot find session in session manager, got nil")
+	}
+
+	// Make sure cookie containing encoded session id is set in response.
+	cookies := rr.HeaderMap["Set-Cookie"]
+	if len(cookies) < 1 {
+		t.Fatal("cannot find cookie response")
+	}
+	esid := encodeSessionID(sess.ID)
```

Comment:
This is testing the internal implementation detail of how this session ID is encoded which isn't very useful. All you really care about is that it can be decoded which is tested in the other unit test, so I would just remove this part.

Reply:
The decoded part is happening in HTTP request. But we want to make sure the session manager successfully sets the "Set-cookie" header in the response with an expected value.